src/GovUk.Education.ExploreEducationStatistics.Admin/ViewModels/EmbedBlockViewModels.cs

```diff
@@ -0,0 +1,27 @@
+#nullable enable
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.ViewModels;
+
+public record EmbedBlockCreateRequest
+{
+    [Required] public string Title { get; init; } = string.Empty;
+
+    [Required]
+    [Url]
+    [RegularExpression(@"^https://department-for-education\.shinyapps\.io/.+$", ErrorMessage="URL not permitted")]
+    public string Url { get; init; } = string.Empty;
+
+    [Required] public Guid ContentSectionId { get; init; }
+}
+
+public record EmbedBlockUpdateRequest
+{
+    [Required] public string Title { get; init; } = string.Empty;
+
+    [Required]
+    [Url]
+    [RegularExpression(@"^https://department-for-education\.shinyapps\.io/.+$", ErrorMessage="URL not permitted")]
```

Comment:
To save repeating ourselves, we could refactor this so that the pattern is re-used from a constant string e.g.

```cs
public record EmbedBlockCreateRequest
{
    public const string UrlPattern = @"^https://department-for-education\.shinyapps\.io/.+$";
}

public record EmbedBlockUpdateRequest
{
    [RegularExpression(EmbedBlockCreateRequest.UrlPattern, ErrorMessage="URL not permitted")]
    public string Url { get; init; } = string.Empty;
}
```

Reply:
I had to wrap the const variable in a static class so it can be accessed by both records:
```
public static class UrlPattern
{
    public const string Pattern = @"^https://department-for-education\.shinyapps\.io/.+$";
}
```

Reply:
My bad - missed the `EmbedBlockCreateRequest.UrlPattern`.